### PR TITLE
Sample CR for external dns operator to work on Azure in Openshift.

### PR DESCRIPTION
**Pull request by miheer**
Sample CR for external dns operator to work on Azure in Openshift.

### CODE REVIEW

File: config/samples/azure/azure_openshift_route.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    matchType: Exact
+    name: test-azure1.qe.azure.devcluster.openshift.com
+  provider:
+    type: Azure
+  source:
+    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
+    # so that the external dns will use the router canonical name correrponding to it to create a dns record.
+    openshiftRouteOptions:
+      routerName: default
+    # Source Type is route resource of OpenShift
```
Comment (alebedev87) [inline config/samples/azure/azure_openshift_route.yaml:17]: Where is `type` field?

File: config/samples/azure/kustomization.yaml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 resources:
+- azure_openshift_route.yaml
```
Comment (alebedev87) [inline config/samples/azure/kustomization.yaml:2]: I'm fine with any name but I would really prefer the things to be consistent.
The other file in this directory is called `operator_v1alpha1_externaldns_detailed.yaml`. BTW it's the kubebuilder pattern for the samples.

File: config/samples/azure/azure_openshift_route.yaml (hunk 1 above)
Comment (miheer) [reply config/samples/azure/azure_openshift_route.yaml:17]: line 21

Comment (miheer) [reply config/samples/azure/azure_openshift_route.yaml]: anyway I made the changes

**Comment by miheer** [comment]
@alebedev87 PTAL. Thanks!

**Pull request merged**